pyfa_tool/modules/reading_fa.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Jan 10 09:04:32 2024
+
+@author: thoverga
+"""
+
+
+import os
+import sys
+import xarray as xr
+import numpy as np
+import rioxarray
+from datetime import datetime, timedelta
+
+import pyfa_tool.modules.IO as IO
+import pyfa_tool.modules.geospatial_functions as geospatial_func
+# =============================================================================
+# Formatters
+# =============================================================================
+
+def _fmt_basedate(string, fmt="%Y-%m-%d"):
+    return datetime.strptime(string, fmt)
+
+def _fmt_validate(string, fmt="%Y-%m-%d %H:%M:%S"):
```

Comment:
The format here is assumed to have a time as well.
I have an example where this is not the case.

```bash
data.import_fa(whitelist='CLSTEMPERATURE')
```

```error
Reading json data
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/dodrio/scratch/projects/2022_200/project_output/RMIB-UGent/vsc46032_kobe/PyFa/pyfa_tool/dataset.py", line 420, in import_fa
    ds = reading_fa.json_to_full_dataset(jsonfile,
  File "/dodrio/scratch/projects/2022_200/project_output/RMIB-UGent/vsc46032_kobe/PyFa/pyfa_tool/modules/reading_fa.py", line 73, in json_to_full_dataset
    'validate': _fmt_validate(data['pyfa_metadata']['validate'][0]),
  File "/dodrio/scratch/projects/2022_200/project_output/RMIB-UGent/vsc46032_kobe/PyFa/pyfa_tool/modules/reading_fa.py", line 27, in _fmt_validate
    return datetime.strptime(string, fmt)
  File "/readonly/dodrio/apps/RHEL8/zen2-ib/software/Python/3.10.4-GCCcore-11.3.0/lib/python3.10/_strptime.py", line 568, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
  File "/readonly/dodrio/apps/RHEL8/zen2-ib/software/Python/3.10.4-GCCcore-11.3.0/lib/python3.10/_strptime.py", line 349, in _strptime
    raise ValueError("time data %r does not match format %r" %
ValueError: time data '2011-01-13' does not match format '%Y-%m-%d %H:%M:%S'
```

The FaFile object metadata property:
```bash
 FA.metadata
{'basedate': ['2011-01-01'], 'validate': ['2011-01-13'], 'leadtime': ['288'], 'timestep': ['300'], 'origin': ['tests/data/ICMSHABOF+0288'], 'projection': ['lcc'], 'lon_0': [9.9], 'lat_1': [49], 'lat_2': [49], 'proj_R': [6371229], 'nx': [499], 'ny': [499], 'dx': [12500], 'dy': [12500], 'ex': [13], 'ey': [13], 'center_lon': [9.9], 'center_lat': [49], 'nfields': [410], 'filepath': ['tests/data/ICMSHABOF+0288'], 'ndlux': [499], 'ndgux': [499], 'nsmax': [255], 'nlev': [46], 'refpressure': [101325], 'A_list': [0, 0.0001, 0.0003, 0.0007, 0.0013, 0.0023, 0.0045, 0.0051, 0.0073, 0.0092, 0.0108, 0.0123, 0.0135, 0.0145, 0.0154, 0.016, 0.0165, 0.0168, 0.0169, 0.0169, 0.0168, 0.0165, 0.0161, 0.0157, 0.0151, 0.0144, 0.0137, 0.0129, 0.0121, 0.0112, 0.0103, 0.0094, 0.0085, 0.0075, 0.0066, 0.0057, 0.0048, 0.004, 0.0032, 0.0025, 0.0018, 0.0012, 0.0008, 0.0004, 0.0001, 0, 0], 'B_list': [0, 0, 0, 0, 0, 0, 0, 0.0035, 0.0102, 0.0201, 0.0328, 0.0484, 0.0666, 0.0871, 0.11, 0.1349, 0.1617, 0.1903, 0.2204, 0.2519, 0.2846, 0.3184, 0.3531, 0.3884, 0.4243, 0.4606, 0.497, 0.5335, 0.5698, 0.6057, 0.6412, 0.676, 0.7099, 0.7429, 0.7746, 0.805, 0.8338, 0.8609, 0.8862, 0.9094, 0.9304, 0.949, 0.965, 0.9783, 0.9887, 0.996, 1]}
```

It seems to have a lead time and timestep that would still have to be translated to a valid time somewhere?

Reply:
@kobebryant432, are you going to fix this or should I ?

Reply:
Its a big file (200mb) but will do. See fix in new branch :)

Reply:
Its to big, it does not want to upload the file.